### PR TITLE
Vitest

### PR DESCRIPTION
**Pull request by shravan-cc**
Implemented testing on linked list functions by making use of vitest. Performed functions such as removing an element, inserting element before and after a particular node on a list, creating an array from list and  filtering array from the list. Completed all the given tasks.

### CODE REVIEW

File: linked-list.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
+import assert from "assert";
+
+export function createList(array = []) {
```
Comment (shripada) [inline linked-list.js]: naming this `array` is confusing, can call it as `arrayOrList` instead.  Also de

**Review by shripada — CHANGES_REQUESTED** [review]

File: linked-list.js (hunk 1 above)
Comment (shravan-cc) [reply linked-list.js]: Implemented the changes.

**Review by shripada — CHANGES_REQUESTED** [review]
why node_modules/ content are getting added to repo?

**Review by shripada — CHANGES_REQUESTED** [review]
Also eslint.config.js is missing ?

File: linked-list.js [hunk 2]
```diff
@@ -0,0 +1,399 @@
+/* eslint-disable consistent-return */
```
Comment (shripada) [inline linked-list.js]: why are we requiring to disable these?

File: linked-list.js [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+    return list;
+  }
+  return { head: null, tail: null };
+}
+
+/**
+ * adds data into a fresh node towards the end of the list
+ */
+
+export function addItemToList(listRef, data) {
```
Comment (shripada) [inline linked-list.js:53]: jsdoc should be written in full manner

File: linked-list.js [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+  // assert(listRef !== undefined && listRef !== null);
```
Comment (shripada) [inline linked-list.js]: why pre conditions are commented out?

File: linked-list.js [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+    listRef.tail = null;
+    // eslint-disable-next-line consistent-return
+    return;
+  }
+  let prevNode = null;
+  let lastNode = listRef.head;
+  while (lastNode !== null) {
+    if (lastNode.data === data) {
+      prevNode.next = lastNode.next;
+      // eslint-disable-next-line consistent-return
```
Comment (shripada) [inline linked-list.js]: We need to stop disabling eslint like this.  Guess such disabling only defeats the purpose of ESLint.

File: linked-list.js [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+    let lastNode = this.head;
+    while (lastNode !== null) {
+      if (predicate(lastNode.data)) {
+        array.push(lastNode.data);
+      }
+      lastNode = lastNode.next;
+    }
+    return array;
+  }
+  traverse(visit) {
```
Comment (shripada) [inline linked-list.js:397]: need to leave blank space between implementation of two functions

**Review by shripada — CHANGES_REQUESTED** [review]

File: linked-list.js (hunk 4 above)
Comment (shravan-cc) [reply linked-list.js]: Implemented the changes

File: linked-list.js (hunk 2 above)
Comment (shravan-cc) [reply linked-list.js]: Implemented the changes

File: linked-list.js (hunk 6 above)
Comment (shravan-cc) [reply linked-list.js:397]: Implemented the changes.

File: linked-list.js (hunk 5 above)
Comment (shravan-cc) [reply linked-list.js]: Implemented the changes and modified the code by not disabling it.

**Pull request merged**